avocado/utils/datadrainer.py

```diff
@@ -0,0 +1,194 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2019
+# Author: Cleber Rosa <crosa@redhat.com>
+
+"""
+data drainer
+
+This module provides utility classes for draining data and dispatching
+it to different destinations.  This is intended to be used
+concurrenly with other code, usually test code producing the output
+to be drained/processed.  A thread is started and maintained on behalf
+of the user.
+"""
+
+import abc
+import io
+import os
+import select
+import threading
+
+
+class BaseDrainer:
+
+    __metaclass__ = abc.ABCMeta
```

Comment:
Almost there :)
This syntax is python2. In python3, you have to use `class BaseDrainer(metaclass=abc.ABCMeta):` instead.
Sorry, I should've been more descriptive to begin with.

Reply:
It's my bad, I did that in a rush, and didn't do a negative test.  This actually reveals that `avocado/core/plugin_interfaces.py` also needs an update.